### PR TITLE
import type { PyProxy } from "pyodide/ffi" doesn't work with NodeNext…

### PR DESCRIPTION
**Pull request by quassnoi**
… (#4255)

### Description

```
import type { PyProxy } from "pyodide/ffi"
```

 doesn't work with NodeNext, this PR fixes it.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

**Comment by hoodmane** [comment]
Thanks @quassnoi. we attempted to cover this with the tests here:
https://github.com/pyodide/pyodide/blob/main/src/test-js/index.test-d.ts

I'll try to look into why that test failed to catch it and see if we can fix that as part of this PR.

**Comment by quassnoi** [comment]
> Thanks @quassnoi. we attempted to cover this with the tests here: https://github.com/pyodide/pyodide/blob/main/src/test-js/index.test-d.ts
> 
> I'll try to look into why that test failed to catch it and see if we can fix that as part of this PR.

If there's an easy way to run just the tests on the JS part, let me know how to do that and I can take look at it tonight.

I was trying to run `make`, but after a half an hour's battle with automake, I gave up and automake won.

**Comment by hoodmane** [comment]
Okay, this needs the merge changelog conflict resolved, but other than that it should hopefully be good to go.

File: docs/project/changelog.md [hunk 1]
```diff
@@ -28,6 +28,9 @@ myst:
   `loadPyodide`, but this may be removed in the future.
   {pr}`4247`
 
+- {{ Fix }} `import type { PyProxy } from "pyodide/ffi"` doesn't work with NodeNext.
```
Comment (hoodmane) [inline docs/project/changelog.md:31]: Maybe:
```suggestion
- {{ Fix }} `import type { PyProxy } from "pyodide/ffi"` now works with the `NodeNext` typescript target.
```

**Comment by hoodmane** [comment]
Thanks @quassnoi!

**Pull request merged**